src/transactions/OperationFrame.h

```diff
@@ -57,39 +65,35 @@ class OperationFrame
     virtual bool isOpSupported(LedgerHeader const& header) const;
 
     LedgerTxnEntry loadSourceAccount(AbstractLedgerTxn& ltx,
-                                     LedgerTxnHeader const& header);
-
-    // given an operation, gives a default value representing "success" for the
-    // result
-    void resetResultSuccess();
+                                     LedgerTxnHeader const& header) const;
 
   public:
     static std::shared_ptr<OperationFrame>
-    makeHelper(Operation const& op, OperationResult& res,
-               TransactionFrame& parentTx, uint32_t index);
+    makeHelper(Operation const& op, TransactionFrame const& parentTx,
+               uint32_t index);
 
-    OperationFrame(Operation const& op, OperationResult& res,
-                   TransactionFrame& parentTx);
+    OperationFrame(Operation const& op, TransactionFrame const& parentTx);
     OperationFrame(OperationFrame const&) = delete;
     virtual ~OperationFrame() = default;
 
+    // given an operation result, gives a default value representing "success"
+    void resetResultSuccess(OperationResult& res) const;
```

Comment:
Can we make this a 'factory' method for consistency with tx frame result? I.e. `OperationResult createSuccessResult() const`.

Reply:
No. `OperationResult` is stored in a vector inside of `TransactionResult`. We can't create a new `TransactionResult` here because this gets called on a per-op instead of per-tx basis. It's also difficult to create a new OperationResult object and move it into the vector where the old result used to be because it's an XDR type, not a true C++ type, so I think taking a reference here is the cleanest approach.